src/Psl/Collection/MutableSet.php

```diff
@@ -0,0 +1,716 @@
+     *
+     * @psalm-mutation-free
+     */
+    public function chunk(int $size): MutableVector
+    {
+        /**
+         * @psalm-suppress MissingThrowsDocblock
+         * @psalm-suppress ImpureFunctionCall
+         */
+        return MutableVector::fromArray(Vec\map(
+            /**
+             * @psalm-suppress MissingThrowsDocblock
+             * @psalm-suppress ImpureFunctionCall
+             */
+            Vec\chunk($this->toArray(), $size),
+            /**
+             * @param list<T> $chunk
+             *
+             * @return MutableSet<T>
+             */
+            static fn(array $chunk) => MutableSet::fromArrayKeys($chunk)
+        ));
+    }
+
+    /**
+     * Determines if the specified offset exists in the current set.
+     *
+     * @param mixed $offset An offset to check for.
+     *
+     * @throws Exception\InvalidOffsetException If the offset type is not valid.
+     *
+     * @return bool Returns true if the specified offset exists, false otherwise.
+     *
+     * @psalm-mutation-free
+     *
+     * @psalm-assert array-key $offset
+     */
+    public function offsetExists(mixed $offset): bool
+    {
+        if (!is_int($offset) && !is_string($offset)) {
+            throw new Exception\InvalidOffsetException('Invalid set read offset type, expected a string or an integer.');
+        }
+
+        /** @var T $offset - technically, we don't know if the offset is of type T, but we can assume it is, as this causes no "harm". */
+        return $this->contains($offset);
+    }
+
+    /**
+     * Returns the value at the specified offset.
+     *
+     * @param mixed $offset The offset to retrieve.
+     *
+     * @throws Exception\InvalidOffsetException If the offset type is not array-key.
+     * @throws Exception\OutOfBoundsException If the offset does not exist.
+     *
+     * @return T The value at the specified offset.
+     *
+     * @psalm-mutation-free
+     *
+     * @psalm-assert array-key $offset
+     */
+    public function offsetGet(mixed $offset): mixed
+    {
+        if (!is_int($offset) && !is_string($offset)) {
+            throw new Exception\InvalidOffsetException('Invalid set read offset type, expected a string or an integer.');
+        }
+
+        /** @var T $offset - technically, we don't know if the offset is of type T, but we can assume it is, as this causes no "harm". */
+        return $this->at($offset);
+    }
+
+    /**
+     * Sets the value at the specified offset.
+     *
+     * @param mixed $offset The offset to assign the value to.
+     * @param T $value The value to set.
+     *
+     * @psalm-external-mutation-free
+     *
+     * @psalm-assert null|array-key $offset
+     *
+     * @throws Exception\InvalidOffsetException If the offset is not null or the value is not the same as the offset.
+     */
+    public function offsetSet(mixed $offset, mixed $value): void
+    {
+        if (null === $offset || $offset === $value) {
```

Comment:
Per the earlier discussion about keys not really being a valid concept of sets, we might even consider disallowing the use of non-null offsets, so `$mutableSet[] = 123` is legal but `$mutableSet[123] = ...` is illegal, regardless of what `...` is.

Reply:
I think it is okay to keep support for this, it does not cause any problems here.